Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/AssetBrowserThumbnailViewProxyModel.cpp

```diff
@@ -115,5 +120,109 @@ namespace AzToolsFramework
+        {
+            return Qt::CopyAction | Qt::MoveAction;
+        }
+
+        Qt::ItemFlags AssetBrowserThumbnailViewProxyModel::flags(const QModelIndex& index) const
+        {
+            Qt::ItemFlags defaultFlags = QAbstractItemModel::flags(index) | Qt::ItemIsEditable;
+
+            if (index.isValid())
+            {
+                // We can only drop items onto folders so set flags accordingly
+                const AssetBrowserEntry* item = mapToSource(index).data(AssetBrowserModel::Roles::EntryRole).value<const AssetBrowserEntry*>();
+                if (item)
+                {
+                    if (item->RTTI_IsTypeOf(ProductAssetBrowserEntry::RTTI_Type()) || item->RTTI_IsTypeOf(SourceAssetBrowserEntry::RTTI_Type()))
+                    {
+                        return Qt::ItemIsDragEnabled | defaultFlags;
+                    }
+                    if (item->RTTI_IsTypeOf(FolderAssetBrowserEntry::RTTI_Type()))
+                    {
+                        return Qt::ItemIsDragEnabled | Qt::ItemIsDropEnabled | defaultFlags;
+                    }
+                }
+            }
+            return defaultFlags;
+        }
+
+        bool AssetBrowserThumbnailViewProxyModel::dropMimeData(const QMimeData* data, Qt::DropAction action, int row, int column, const QModelIndex& parent)
+        {
+            if (action == Qt::IgnoreAction)
+            {
+                return true;
+            }
+
+            auto sourceparent = mapToSource(parent).data(AssetBrowserModel::Roles::EntryRole).value<const AssetBrowserEntry*>();
+
+            // We should only have an item as a folder but will check
+            if (sourceparent && (sourceparent->RTTI_IsTypeOf(FolderAssetBrowserEntry::RTTI_Type())))
+            {
+                AZStd::vector<const AssetBrowserEntry*> entries;
+
+                if (Utils::FromMimeData(data, entries))
+                {
+                    for (auto entry : entries)
+                    {
+                        using namespace AZ::IO;
+                        Path fromPath;
+                        Path toPath;
+                        bool isFolder{ true };
+
+                        if (entry && (entry->RTTI_IsTypeOf(SourceAssetBrowserEntry::RTTI_Type())))
+                        {
+                            fromPath = entry->GetFullPath();
+                            PathView filename = fromPath.Filename();
+                            toPath = sourceparent->GetFullPath();
+                            toPath /= filename;
+                            isFolder = false;
+                        }
+                        else
+                        {
+                            fromPath = entry->GetFullPath() + "/*";
+                            Path filename = static_cast<Path>(entry->GetFullPath()).Filename();
+                            toPath = sourceparent->GetFullPath() + "/" + filename.c_str() + "/*";
```

Comment:
Nit: If you use AZ::IO::Path here instead of the full path string, it would make this logic a little cleaner.

Reply:
Done